@mrinalwadhwa I've finished the implementation for the mentioned issue.
While playing with the app, I realized it stops when I run `reset` command.
This is the error(it's not related to this change, it also happens on `develop`):
```
Local Ockam configuration deleted
[2023-07-19][12:23:01][ockam_app_lib::enroll::menu_items][INFO] There is no default project, please enroll
[2023-07-19][12:23:01][ockam_node::node][INFO] Initializing ockam node
[2023-07-19][12:23:01][mio::poll][TRACE] registering event source with poller: token=Token(2147483649), interests=READABLE
[2023-07-19][12:23:01][ockam_node::executor][TRACE] Initializing node executor
[2023-07-19][12:23:01][ockam_node::router::start_worker][DEBUG] Starting new worker '0#205492460f2276cea3d4f385f5532eb7'
[2023-07-19][12:23:01][ockam_node::router::start_worker][DEBUG] Starting new worker '0#61e38e462f5a32312be46ca11063802e'
[2023-07-19][12:23:01][ockam_node::router::start_worker][DEBUG] Starting new worker '0#609af5b9d3f21472847fc031c2703242'
[2023-07-19][12:23:01][ockam_command::node::util][DEBUG] initializing node state name=default
[2023-07-19][12:23:01][ockam_node::context::context_lifecycle][TRACE] De-allocated detached context 0#61e38e462f5a32312be46ca11063802e
[2023-07-19][12:23:01][ockam_node::context::context_lifecycle][TRACE] De-allocated detached context 0#205492460f2276cea3d4f385f5532eb7
[2023-07-19][12:23:01][ockam_command::util][ERROR] Failed to run command e=No such file or directory (os error 2)
[2023-07-19][12:23:01][ockam_node::async_drop][DEBUG] Received AsyncDrop request for address: 0#61e38e462f5a32312be46ca11063802e
[2023-07-19][12:23:01][ockam_node::router::stop_worker][TRACE] Stopping worker '0#61e38e462f5a32312be46ca11063802e'
OCK500 (link)

[2023-07-19][12:23:01][ockam_core::flow_control::flow_controls::flow_controls_cleanup][DEBUG] Cleanup FlowControls for 0#61e38e462f5a32312be46ca11063802e
  × No such file or directory (os error 2)
```

--- a/implementations/rust/ockam/ockam_app/src/app/process.rs
+++ b/implementations/rust/ockam/ockam_app/src/app/process.rs
@@ -8,6 +8,7 @@ pub fn process_system_tray_event(app: &AppHandle<Wry>, event: SystemTrayEvent) {
     if let SystemTrayEvent::MenuItemClick { id, .. } = event {
         let result = match id.as_str() {
             enroll::ENROLL_MENU_ID => enroll::on_enroll(app),
+            enroll::PLEASE_ENROLL_MENU_ID => enroll::on_enroll(app),
             tcp::outlet::TCP_OUTLET_CREATE_MENU_ID => tcp::outlet::on_create(app),
             enroll::RESET_MENU_ID => enroll::on_reset(app),
             quit::QUIT_MENU_ID => quit::on_quit(app),

--- a/implementations/rust/ockam/ockam_app/src/app/system_tray.rs
+++ b/implementations/rust/ockam/ockam_app/src/app/system_tray.rs
@@ -1,3 +1,4 @@
+use ockam_api::cli_state::StateDirTrait;
 use tauri::{AppHandle, CustomMenuItem, SystemTray, SystemTrayMenu, SystemTrayMenuItem, Wry};
 use tracing::log::{error, info};
 
@@ -24,7 +25,7 @@ pub struct SystemTrayMenuBuilder {
 impl SystemTrayMenuBuilder {
     /// Create the default system tray menu with the basic elements (i.e. without list items).
     pub fn default(options: &CommandGlobalOpts) -> SystemTrayMenu {
-        Self::init(options).build()
+        Self::init(options).build(options)
     }
 
     pub fn init(options: &CommandGlobalOpts) -> Self {
@@ -35,13 +36,30 @@ impl SystemTrayMenuBuilder {
     }
 
     /// Create a `SystemTrayMenu` instance, adding the menu items in the expected order.
-    pub fn build(self) -> SystemTrayMenu {
+    pub fn build(self, options: &CommandGlobalOpts) -> SystemTrayMenu {
+        match options.state.projects.default() {
+            Ok(_) => Self::init(options).after_enroll(),
+            Err(_) => Self::init(options).before_enroll(),
+        }
+    }
+
+    fn after_enroll(self) -> SystemTrayMenu {
         SystemTrayMenu::new()
             .add_menu_items(&[self.enroll.enroll])
+            .add_menu_items(&[self.enroll.please_enroll])
             .add_native_item(SystemTrayMenuItem::Separator)
             .add_menu_items(&self.tcp.menu_items)
             .add_native_item(SystemTrayMenuItem::Separator)
             .add_menu_items(&[self.enroll.reset, self.quit.quit])
+    }
+
+    fn before_enroll(self) -> SystemTrayMenu {
+        SystemTrayMenu::new()
+            .add_menu_items(&[self.enroll.enroll])
+            .add_native_item(SystemTrayMenuItem::Separator)
+            .add_menu_items(&[self.enroll.please_enroll])
+            .add_native_item(SystemTrayMenuItem::Separator)
+            .add_menu_items(&[self.quit.quit])
     }
 
     /// Refresh the system tray menu with the latest state, including all list items.
@@ -62,7 +80,7 @@ impl SystemTrayMenuBuilder {
         let enroll = EnrollActions::new(options);
         let tcp = TcpOutletActions::full(app, options).await?;
         let quit = QuitActions::new();
-        let menu = Self { enroll, tcp, quit }.build();
+        let menu = Self { enroll, tcp, quit }.build(options);
         Ok(menu)
     }
 }

--- a/implementations/rust/ockam/ockam_app/src/enroll/menu_items.rs
+++ b/implementations/rust/ockam/ockam_app/src/enroll/menu_items.rs
@@ -1,5 +1,5 @@
 use tauri::Error::Runtime;
-use tauri::{AppHandle, CustomMenuItem, Wry};
+use tauri::{AppHandle, CustomMenuItem, Manager, Wry};
 use tauri_runtime::Error::SystemTray;
 use tracing::info;
 
@@ -10,23 +10,28 @@ use crate::enroll::enroll_user::enroll_user;
 use crate::enroll::reset::reset;
 
 pub const ENROLL_MENU_ID: &str = "enroll";
+pub const PLEASE_ENROLL_MENU_ID: &str = "please_enroll";
 pub const RESET_MENU_ID: &str = "reset";
 
 #[derive(Clone)]
 pub struct EnrollActions {
     pub options: CommandGlobalOpts,
     pub(crate) enroll: CustomMenuItem,
+    pub(crate) please_enroll: CustomMenuItem,
     pub(crate) reset: CustomMenuItem,
 }
 
 impl EnrollActions {
     pub fn new(options: &CommandGlobalOpts) -> EnrollActions {
-        let enroll = CustomMenuItem::new(ENROLL_MENU_ID, "Enroll").accelerator("cmd+e");
+        let enroll = CustomMenuItem::new(ENROLL_MENU_ID, "Enroll...").accelerator("cmd+e");
+        let please_enroll =
+            CustomMenuItem::new(PLEASE_ENROLL_MENU_ID, "Please Enroll.").accelerator("cmd+e");
         let reset = CustomMenuItem::new(RESET_MENU_ID, "Reset").accelerator("cmd+r");
         match options.state.projects.default() {
             Ok(_) => EnrollActions {
                 options: options.clone(),
                 enroll: enroll.disabled(),
+                please_enroll: please_enroll.disabled(),
                 reset,
             },
             Err(_) => {
@@ -34,6 +39,7 @@ impl EnrollActions {
                 EnrollActions {
                     options: options.clone(),
                     enroll,
+                    please_enroll,
                     reset: reset.disabled(),
                 }
             }
@@ -45,6 +51,7 @@ impl EnrollActions {
 pub fn on_enroll(app: &AppHandle<Wry>) -> tauri::Result<()> {
     let app_handle = app.clone();
     tauri::async_runtime::spawn(async move { enroll_user(&app_handle).await });
+    app.trigger_global(crate::app::events::SYSTEM_TRAY_ON_UPDATE, None);
     Ok(())
 }
 

